code/defines/obj/weapon.dm

```diff
@@ -138,10 +138,78 @@
 
 /obj/item/weapon/cane/whitecane
 	name = "white cane"
-	desc = "A cane used by the blind."
+	desc = "A white cane. They are commonly used by the blind or visually impaired as a mobility tool or as a courtesy to others."
 	icon = 'icons/obj/weapons.dmi'
 	icon_state = "whitecane"
 
+/obj/item/weapon/cane/whitecane/attack(mob/M as mob, mob/user as mob)
+    if(user.a_intent == I_HELP)
+        user.visible_message("<span class='notice'>\The [user] has lightly tapped [M] on the ankle with their white cane!</span>")
+        return
+    else
+        ..()
+
+/obj/item/weapon/cane/crutch
+	name ="crutch"
+	desc = "A long stick with a crosspiece at the top, used to help with walking."
+	icon_state = "crutch"
+	item_state = "crutch"
+
+//Code for Telescopic White Cane writen by Gozulio
+
+/obj/item/weapon/melee/collapsable_whitecane
+	name = "telescopic white cane"
+	desc = "A telescoping white cane. They are commonly used by the blind or visually impaired as a mobility tool or as a courtesy to others."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "whitecane1in"
+	item_icons = list(
+		slot_l_hand_str = 'icons/mob/items/lefthand_melee.dmi',
+		slot_r_hand_str = 'icons/mob/items/righthand_melee.dmi',
+	)
+	slot_flags = SLOT_BELT
+	w_class = ITEMSIZE_SMALL
+	force = 3
+	var/on = 0
+
+/obj/item/weapon/melee/collapsable_whitecane/attack_self(mob/user as mob)
+	on = !on
+	if(on)
+		user.visible_message("<span class='notice'>\The [user] extends the white cane.</span>",\
+		"<span class='warning'>You extend the white cane.</span>",\
+		"You hear an ominous click.")
+		icon_state = "whitecane1out"
+		item_state_slots = list(slot_r_hand_str = "whitecane", slot_l_hand_str = "whitecane")
+		w_class = ITEMSIZE_NORMAL
+		force = 5
+		attack_verb = list("smacked", "struck", "cracked", "beaten")
+	else
+		user.visible_message("<span class='notice'>\The [user] collapses the white cane.</span>",\
+		"<span class='notice'>You collapse the white cane.</span>",\
+		"You hear a click.")
+		icon_state = "whitecane1in"
+		item_state_slots = list(slot_r_hand_str = null, slot_l_hand_str = null)
+		w_class = ITEMSIZE_SMALL
+		force = 3
+		attack_verb = list("hit", "poked")
+
+	if(istype(user,/mob/living/carbon/human))
+		var/mob/living/carbon/human/H = user
+		H.update_inv_l_hand()
+		H.update_inv_r_hand()
+
+	playsound(src.loc, 'sound/weapons/empty.ogg', 50, 1)
+	add_fingerprint(user)
+
+	return
+
+/obj/item/weapon/melee/collapsable_whitecane/attack(mob/M as mob, mob/user as mob)
+	if(user.a_intent == I_HELP)

```

Comment:
`if(on && user.a_intent == I_HELP)`
kinda hard to tap someone on the ankle if your telescoping cane isn't extended